src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -646,16 +623,14 @@ private void sendBulkLoadBCP() throws SQLServerException {
         final class InsertBulk extends TDSCommand {
             InsertBulk() {
                 super("InsertBulk", 0, 0);
-                int timeoutSeconds = copyOptions.getBulkCopyTimeout();
-                timeoutCommand = timeoutSeconds > 0 ? new BulkTimeoutCommand(timeoutSeconds, this, null) : null;
             }
 
             final boolean doExecute() throws SQLServerException {
-                if (null != timeoutCommand) {
-                    if (logger.isLoggable(Level.FINEST))
-                        logger.finest(this.toString() + ": Starting bulk timer...");
-
-                    TimeoutPoller.getTimeoutPoller().addTimeoutCommand(timeoutCommand);
+                int timeoutSeconds = copyOptions.getBulkCopyTimeout();
+                if (timeoutSeconds > 0) {
+                    connection.checkClosed();
```

Comment:
Is this check required here? This code path is executed after the check has passed.

Reply:
Yes, the previous code had that check done in a ternary in the constructor.

My last note in the first description of this PR references this:

> The clock for cancellation of bulk inserts now starts when the bulk command is going to execute as opposed to when it is created. The only blocking operation in between is waiting on an internal synchronization monitor, schedulerLock, so in practice it'd be microseconds of difference.